mozci/console/commands/push.py

```diff
@@ -946,6 +955,107 @@ def list_classification_tasks(self, group_id):
             config.cache.add(cache_key, tasks, int(config["cache"]["retention"]))
 
 
+class CheckBackfillsCommand(Command):
+    """
+    Check if backfills on last pushes are finished and notify Sheriffs when they are.
+
+    check-backfills
+        {--branch=autoland : Branch the pushes belongs to (e.g autoland, try, etc).}
+        {--nb-pushes=100 : Number of recent pushes to retrieve for the check.}
+        {--environment=testing : Environment in which the analysis is running (testing, production, ...)}
+    """
+
+    def handle(self) -> None:
+        branch = self.option("branch")
+        environment = self.option("environment")
+        matrix_room = config.get("matrix-room-id", None)
+
+        try:
+            nb_pushes = int(self.option("nb-pushes"))
+        except ValueError:
+            self.line("<error>Provided --nb-pushes should be an int.</error>")
+            exit(1)
+
+        self.line("<comment>Loading pushes...</comment>")
+        self.pushes = make_push_objects(nb=nb_pushes, branch=branch)
+
+        for push in self.pushes:
+            backfill_tasks = []
+
+            try:
+                indexed_tasks = list_indexed_tasks(
+                    f"gecko.v2.{push.branch}.revision.{push.rev}.taskgraph.actions"
+                )
+            except requests.exceptions.HTTPError as e:
+                self.line(
+                    f"<error>Couldn't fetch indexed tasks on push {push.push_uuid}: {e}</error>"
+                )
+                continue
+
+            for indexed_task in indexed_tasks:
+                task_id = indexed_task["taskId"]
+                try:
+                    children_tasks = list_dependent_tasks(task_id)
+                except requests.exceptions.HTTPError as e:
+                    self.line(
+                        f"<error>Couldn't fetch dependent tasks of indexed task {task_id} on push {push.push_uuid}: {e}</error>"
+                    )
+                    continue
+
+                for child_task in children_tasks:
+                    task_action = (
+                        child_task.get("task", {}).get("tags", {}).get("action", "")
+                    )
+                    th_symbol = (
+                        child_task.get("task", {})
+                        .get("extra", {})
+                        .get("treeherder", {})
+                        .get("symbol", "")
+                    )
+                    if task_action == "backfill-task" and th_symbol.endswith("-bk"):
+                        status = child_task.get("status", {})
+                        backfill_tasks.append(
+                            {
+                                "taskId": status.get("taskId"),
```

Comment:
I would prefer a namedtuple here, and stricter checks: the workflow cannot work if any of these information is missing.

So raise an exception when a part is missing